lib/src/report/template.dart

```diff
@@ -251,13 +252,61 @@ Future<ReportSection> followsTemplate(PackageContext context) async {
         'See the [example](https://raw.githubusercontent.com/dart-lang/stagehand/master/templates/package-simple/CHANGELOG.md) generated by `stagehand`.',
   );
   final pubspecSection = await checkPubspec();
-  final subsections = [pubspecSection, readmeSubsection, changelogSubsection];
+  final subsections = [
+    pubspecSection,
+    readmeSubsection,
+    changelogSubsection,
+    _licenseSection(await context.licenceTags),
+  ];
   return makeSection(
     id: ReportSectionId.convention,
     title: 'Follow Dart file conventions',
-    maxPoints: 20,
+    maxPoints: 30,
     subsections: subsections,
     basePath: packageDir,
     maxIssues: 10,
   );
 }
+
+Subsection _licenseSection(LicenseTags tags) {
+  const title = 'Use an OSI-approved license';
+  late List<Paragraph> paragraphs;
+  final maxPoints = 10;
+  var grantedPoints = 0;
+  var reportStatus = ReportStatus.failed;
+
+  String licensePluralized(List value) =>
+      value.length > 1 ? 'licenses' : 'license';
+  String licenseList(List<License> values) =>
+      values.map((e) => '`${e.spdxIdentifier}`').join(', ');
+  final detected = RawParagraph(
+      'Detected ${licensePluralized(tags.licenses)}: ${licenseList(tags.licenses)}.');
+
+  if (tags.isOnlyOsiApproved) {
+    paragraphs = [detected];
+    grantedPoints = maxPoints;
+    reportStatus = ReportStatus.passed;
+  } else if (tags.wasDetected) {
+    paragraphs = [
+      detected,
+      Issue(
+        'The following ${licensePluralized(tags.nonOsiApprovedLicenses)} are not OSI-approved: ${licenseList(tags.nonOsiApprovedLicenses)}.',
+        suggestion:
+            'Consider using an [OSI-approved license](https://opensource.org/licenses)'
+            ' in the `LICENSE` file to share with the community.',
+      ),
+    ];
+    grantedPoints = maxPoints ~/ 2;
+    reportStatus = ReportStatus.partial;
+  } else {
+    paragraphs = [
+      Issue(
+        'No license was recognized.',
+        suggestion:
+            'Consider using an [OSI-approved license](https://opensource.org/licenses)'
+            ' in the `LICENSE` file to share with the community.',
```

Comment:
Shouldn't this hint that we can recognize most SPDX licenses?

Reply:
maybe it's fine... saying SPDX is also a bit of problem if you are using this as advice for finding a license.